src/ocean/util/app/ext/ReopenableFilesExt.d

```diff
@@ -70,14 +94,54 @@ public class ReopenableFilesExt : IApplicationExtension, ISignalExtExtension
 
     ***************************************************************************/
 
-    public this ( SignalExt signal_ext = null, int reopen_signal = SIGHUP )
+    public this ( SignalExt signal_ext = null,
+        int reopen_signal = SIGHUP )
+    {
+        this(signal_ext, null, reopen_signal);
+    }
+
+
+    /***************************************************************************
+
+        Constructor. Optionally registers this extension with the signal
+        extension and/or unix socket extension, activates the handling of the
+        specified signal, which will cause the registered files to be reopened,
+        and activates the handling of the specified unix socket command, which
+        will cause the registered file to be reopened.
+
+        Params:
+            signal_ext = optional SignalExt instance to register with
+            unix_socket_ext = optional UnixSocketExt instance to register with
+            reopen_signal = if signal_ext is non-null, signal to trigger
+                reopening of registered files
+            reopen_command = if unix socket ext is non-null, command to trigger
+                reopenning of the registered file (passed as the first
+                argument to the command).
+
+    ***************************************************************************/
+
+    public this ( SignalExt signal_ext,
+        UnixSocketExt unix_socket_ext,
+        int reopen_signal = SIGHUP,
+        cstring reopen_command = "reopen_logfile")
     {
+        this.cwd = Environment.directory();
+
+        // Environment.directory makes sure for the trailing /, confirm it
+        verify (this.cwd[$-1] == '/');
```

Comment:
Extra space

Reply:
Also the comment sounds weird to me

Reply:
Yes, should be `Environment.directory should always return trailing /, confirm that that's the case`